extensions/amp-video-service/0.1/amp-video-service.js

```diff
@@ -76,14 +148,159 @@ export class VideoService {
+    this.service_ = videoService;
+
+    /** @private @const{!../../../src/video-interface.VideoInterface} */
+    this.video_ = video;
+
+    /** @visibleForTesting {boolean} */
+    this.isPlaying = false;
+  }
+
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   * @param {!VideoService} videoService
+   * @param {!../../../src/video-interface.VideoInterface} video
+   */
+  static create(ampdoc, videoService, video) {
+    return new VideoEntry(ampdoc, videoService, video);
+  }
+
+  /** @param {function()} handler */
+  onPlaybackTick(handler) {
+    this.service_.onTick(() => {
+      if (!this.isPlaying) {
+        return;
+      }
+      handler();
+    });
+  }
+
+  /** */
+  install() {
+    const {element} = this.video_;
+    const signals = element.signals();
+
+    element.dispatchCustomEvent(VideoEvents.REGISTERED);
+
+    // Unlike events, signals are permanent. We can wait for `REGISTERED` at any
+    // moment in the element's lifecycle and the promise will resolve
+    // appropriately each time.
+    signals.signal(VideoEvents.REGISTERED);
+
+    element.whenBuilt()
+        .then(() => this.onBuilt_());
+
+    signals.whenSignal(CommonSignals.LOAD_START)
+        .then(() => this.onLoadStart_());
+  }
+
+  /** @private */
+  onBuilt_() {
+    const {element} = this.video_;
+
+    this.registerCommonActions();
+    this.addEventHandlers_();
+
+    element.classList.add('i-amphtml-video-interface');
+  }
+
+  /** @private */
+  onLoadStart_() {
+    this.maybeTriggerTimeUpdate();
+  }
+
+  /** @private */
+  addEventHandlers_() {
+    const {element} = this.video_;
+
+    listen(element, VideoEvents.PAUSE, () => {
+      this.isPlaying = false;
+    });
+
+    listen(element, VideoEvents.PLAYING, () => {
+      this.isPlaying = true;
+    });
+  }
+
+  /**
+   * Register common actions such as play, pause, etc... so they can be called
+   * using AMP Actions, e.g.: `<button on="tap:myVideo.play">`.
+   * @visibleForTesting
+   */
+  registerCommonActions() {
+    const video = this.video_;
+
+    // Only require ActionTrust.LOW for video actions to defer to platform
+    // specific handling (e.g. user gesture requirement for unmuted playback).
+    const trust = ActionTrust.LOW;
+
+    video.registerAction('play', () => video.play(/* isAuto */ false), trust);
+    video.registerAction('pause', () => video.pause(), trust);
+    video.registerAction('mute', () => video.mute(), trust);
+    video.registerAction('unmute', () => video.unmute(), trust);
+    video.registerAction('fullscreen', () => video.fullscreenEnter(), trust);
+  }
+
+  /**
+   * Triggers a LOW-TRUST timeupdate event consumable by AMP actions if
+   * required.
+   * Frequency of this event is controlled by VideoService.onTick() and is
+   * every second for now.
+   * @visibleForTesting
+   */
+  maybeTriggerTimeUpdate() {
+    const {element} = this.video_;
+
+    if (!closestBySelector(element, '[on*="timeUpdate:"]')) {
```

Comment:
this won't be enough due to chaining support (e.g. `on=tap:foo;timeUpdate:bar"). Dispatching an event every second doesn't seem like a huge perf issue but worth fixing, I just don't think this is the right way. We would want the Event/Action service to do the parsing as it does now and have it expose enough information for cases like this to query what event handlers are installed for a component.

Reply:
sg, removed.